Marlin/src/lcd/extui/lib/nextion/nextion_tft.cpp

```diff
@@ -0,0 +1,726 @@
+      SEND_VALasTXT("tmppage.printer", MACHINE_NAME);
+      SEND_VALasTXT("tmppage.author", STRING_CONFIG_H_AUTHOR);
+      SEND_VALasTXT("tmppage.released", STRING_DISTRIBUTION_DATE);
+      SEND_VALasTXT("tmppage.bedx", X_BED_SIZE);
+      SEND_VALasTXT("tmppage.bedy", Y_BED_SIZE);
+      SEND_VALasTXT("tmppage.bedz", Z_MAX_POS);
+      SEND_TEMP("tmppage.t0", ui8tostr3rj(getActualTemp_celsius(E0)), " / ", ui8tostr3rj(getTargetTemp_celsius(E0)));
+      SEND_TEMP("tmppage.t1", ui8tostr3rj(getActualTemp_celsius(E1)), " / ", ui8tostr3rj(getTargetTemp_celsius(E1)));
+      SEND_TEMP("tmppage.t2", ui8tostr3rj(getActualTemp_celsius(BED)), " / ", ui8tostr3rj(getTargetTemp_celsius(BED)));
+      SEND_VALasTXT("tmppage.tool", getActiveTool());
+      SEND_VALasTXT("tmppage.fan", ui8tostr3rj(getActualFan_percent(FAN0)));
+      SEND_VALasTXT("tmppage.speed", getFeedrate_percent());
+      SEND_VALasTXT("tmppage.flow", getFlowPercentage(getActiveTool()));
+      SEND_VALasTXT("tmppage.progress", ui8tostr3rj(getProgress_percent()));
+      SEND_VALasTXT("tmppage.layer", layer);
+      SEND_VALasTXT("tmppage.x", getAxisPosition_mm(X));
+      SEND_VALasTXT("tmppage.y", getAxisPosition_mm(Y));
+      SEND_VALasTXT("tmppage.z", getAxisPosition_mm(Z));
+      SEND_VAL("tmppage.homed", isPositionKnown());
+      SEND_VAL("tmppage.homedx", isAxisPositionKnown(X));
+      SEND_VAL("tmppage.homedy", isAxisPositionKnown(Y));
+      SEND_VAL("tmppage.homedz", isAxisPositionKnown(Z));
+      #if ENABLED(DUAL_X_CARRIAGE)
+        SEND_VAL("tmppage.idexmode", getIDEX_Mode());
+      #endif
+      SEND_TXT("tmppage.M117", msg_welcome);
+    }
+    break;
+
+  case 23: // Linear Advance
+    #if ENABLED(LIN_ADVANCE)
+      SEND_VALasTXT("linadvance", getLinearAdvance_mm_mm_s(getActiveTool()));
+    #else
+      SEND_NA("linadvance");
+    #endif
+    break;
+
+  case 24: // TMC Motor Current
+    #if HAS_TRINAMIC_CONFIG
+      #define SEND_TRINAMIC_CURR(A, B) SEND_VALasTXT(A, getAxisCurrent_mA(B))
+    #else
+      #define SEND_TRINAMIC_CURR(A, B) SEND_NA(A)
+    #endif
+    SEND_TRINAMIC_CURR("x", X);
+    SEND_TRINAMIC_CURR("x2", X2);
+    SEND_TRINAMIC_CURR("y", Y);
+    SEND_TRINAMIC_CURR("y2", Y2);
+    SEND_TRINAMIC_CURR("z", Z);
+    SEND_TRINAMIC_CURR("z2", Z2);
+    SEND_TRINAMIC_CURR("e", E0);
+    SEND_TRINAMIC_CURR("e1", E1);
+    break;
+
+  case 25: // TMC Bump Sensitivity
+    #if HAS_TRINAMIC_CONFIG
+      #define SEND_TRINAMIC_BUMP(A, B) SEND_VALasTXT(A, getTMCBumpSensitivity(B))
+    #else
+      #define SEND_TRINAMIC_BUMP(A, B) SEND_NA(A)
+    #endif
+    SEND_TRINAMIC_BUMP("x", X);
+    SEND_TRINAMIC_BUMP("x2", X2);
+    SEND_TRINAMIC_BUMP("y", Y);
+    SEND_TRINAMIC_BUMP("y2", Y2);
+    SEND_TRINAMIC_BUMP("z", Z);
+    SEND_TRINAMIC_BUMP("z2", Z2);
+    break;
+
+  case 26: // TMC Hybrid Threshold Speed
+    #if 0 && BOTH(HAS_TRINAMIC_CONFIG, HYBRID_THRESHOLD)
+      #define SEND_TRINAMIC_THRS(A, B) SEND_VALasTXT(A, getAxisPWMthrs(B))
```

Comment:
At the moment there is no accessor for the Hybrid Threshold. So if this section is needed to match up with the display, that will need to be implemented. The code was previously just repeating the calls to `getAxisCurrent_mA`.

Reply:
https://user-images.githubusercontent.com/5186502/111031646-ddc1c300-8408-11eb-9d9f-de88ee60f09b.mov
If this setting makes sense, then it would have to be implemented.

![Bildschirmfoto 2021-03-13 um 14 34 20](https://user-images.githubusercontent.com/5186502/111031743-4f017600-8409-11eb-987d-64043647ca04.png)
Where is the best place to put the code?

`  #define PREHEAT_COUNT 3
  #define PREHEAT_3_LABEL       "PETG"
  #define PREHEAT_3_TEMP_HOTEND 220
  #define PREHEAT_3_TEMP_BED    70
  #define PREHEAT_3_FAN_SPEED   0 // Value from 0 to 255`

Reply:
Hybrid Threshold should be placed with the other Trinamic Stepper Driver settings.

While working on the code, I did see some stuff related to the Preheat items. So any additions to preheating will be added alongside the existing preheat code.